plugins/janus_sip.c

```diff
@@ -2450,18 +2467,32 @@ gpointer janus_sip_sofia_thread(gpointer user_data) {
 		g_strlcat(outbound_options, " options-keepalive", sizeof(outbound_options));
 	if(!behind_nat)
 		g_strlcat(outbound_options, " no-natify", sizeof(outbound_options));
-	session->stack->s_nua = nua_create(session->stack->s_root,
-				janus_sip_sofia_callback,
-				session,
-				SIPTAG_ALLOW_STR("INVITE, ACK, BYE, CANCEL, OPTIONS"),
-				NUTAG_M_USERNAME(session->account.username),
-				NUTAG_URL(sip_url),
-				NUTAG_SIPS_URL(sips_url),
-				SIPTAG_USER_AGENT_STR(user_agent),
-				NUTAG_KEEPALIVE(keepalive_interval * 1000),	/* Sofia expects it in milliseconds */
-				NUTAG_OUTBOUND(outbound_options),
-				SIPTAG_SUPPORTED(NULL),
-				TAG_NULL());
+	if(session->account.sips) {
```

Comment:
I think you can simplify this by using TAG_IF: http://sofia-sip.sourceforge.net/refdocs/su/su__tag_8h.html#a8ccbc9cc9f07d5670065e990d07bb428